R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,702 @@
+    "colnames(result)[1] <- \"Dataset\"\n",
+    "result$Image <- as.integer(result$Image)\n",
+    "\n",
+    "# attach the dataframe to the project\n",
+    "invisible(attachDataframe(project, result, \"Summary from R\"))\n",
+    "\n",
+    "# Change the namespace (for parade) (workaround as this is not yet\n",
+    "# implemented in the R gateway itself)\n",
+    "dataframes <- availableDataframes(project)\n",
+    "faid <- dataframes$AnnotationID[1]\n",
+    "fa <- loadObject(server, \"omero.gateway.model.FileAnnotationData\", as.integer(faid))\n",
+    "fa@dataobject$setNameSpace(\"openmicroscopy.org/omero/bulk_annotations\")\n",
+    "gateway <- getGateway(server)\n",
+    "ctx <- getContext(server)\n",
+    "dm <- gateway$getFacility(DataManagerFacility$class)\n",
+    "fa <- dm$saveAndReturnObject(ctx, fa@dataobject)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load the segmentation results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Instead of running the segmentation over the whole idr0021 project yourself, you'll use the summary table of trainer-1.\n",
+    "\n",
+    "**Exercise:** In OMERO.web got to Lab1 / trainer-1 and find out the ID of the idr0021 project."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "projectId <- REPLACE_WITH_PROJEcT_ID # This is the ID of trainer-1's idr0021 project\n",
+    "project <- loadObject(server, \"ProjectData\", projectId)\n",
+    "dataframes <- availableDataframes(project)\n",
+    "print(dataframes)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the dataframe\n",
+    "dfID <- dataframes$ID[1]\n",
+    "centrioles <- loadDataframe(project, dfID)\n",
+    "\n",
+    "# Make sure the data types are correct\n",
+    "centrioles$Dataset <- as.factor(centrioles$Dataset)\n",
+    "centrioles$Diameter <- as.numeric(centrioles$Diameter)\n",
+    "centrioles"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Plot the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ag <-aggregate(centrioles$Diameter ~ centrioles$Dataset, centrioles, median)\n",
+    "ordered <- factor(centrioles$Dataset, levels=ag[order(ag$`centrioles$Diameter`), 'centrioles$Dataset'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot(centrioles$Diameter ~ ordered, ylab='Diameter (nm)', xlab=\"Protein\", cex.axis=0.5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Find the outlier in OMERO.web (using 'parade') and check what might have gone wrong there."
```

Comment:
``outlier`` -> ``outliers`` 

OMERO.parade will be introduced only later, after this. But, we can highlight this exercise as one to do after the parade is explained.